.github/workflows/test.yml

```diff
@@ -9,33 +9,28 @@ on:
             - main
 
 jobs:
+    set-vars:
+        uses: ./.github/workflows/set-vars.yml
+
     test:
+        needs: [set-vars]
         runs-on: [ubuntu-latest]
         container:
-            image: python:3.9
+            image: '${{ needs.set-vars.outputs.CI_IMAGE }}:latest'
```

Comment:
Are the quotes needed?

Reply:
Also you can get rid of `latest` (this is the default)

Reply:
I added the quotes because it didn't work without the quotes. But now that there's no longer `latest`, it works even without the quotes. So deleted everywhere.